### DISCUSSION TITLE
Plagiarism or similarity checker

### DISCUSSION THREAD

**Discussion by joelostblom** [PrairieLearn/PrairieLearn#7489] [Q&A]
Are there any technologies available to check academic misconduct in students' work, which play nicely with PraireLearn? I am specifically looking for something with the following features:

1. Detecting plagiarism from previous work, e.g. wikipedia, research articles, stack overflow, etc.
2. Detecting similarity between students work to see if students have copy pasted each other's answers
3. Detecting if an AI tool has been used generate the test (this is less important for me right now)
4. A solution that works for both code and free text answers.

I wonder if there are any other options out there that are known to work well with PL. Even just detecting similarity between student work would be very helpful. I haven't used PL much myself yet, but searching in the documentation didn't bring up anything regarding plagiarism for me.

**Comment by eliotwrobson** [comment]
This was attempted in this PR here: https://github.com/PrairieLearn/PrairieLearn/pull/6035

Although this was closed I think this project was restarted. You can ask on slack to discuss with the people working on this.

**Comment by nwalters512** [comment]
I'm going to convert this to a discussion, as this isn't directly a feature request for PrairieLearn itself.
